prober/http.go

```diff
@@ -380,6 +386,12 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 		target = "http://" + target
 	}
 
+	// For HTTP/3, ensure HTTPS is used
+	if httpConfig.UseHTTP3 && strings.HasPrefix(target, "http://") {
+		target = strings.Replace(target, "http://", "https://", 1)
+		logger.Info("Converting HTTP to HTTPS for HTTP/3 compatibility", "target", target)
```

Comment:
this should be at warn level because we are changing the target and users should be aware of this change. not all users run at info level.